core/tasks.py

```diff
@@ -207,7 +208,13 @@ def _score_submission(submission):
     return submission
 
 
-@shared_task(soft_time_limit=60 * 20, time_limit=60 * 21)
+@shared_task(
+    soft_time_limit=60 * 20,
+    time_limit=60 * 21,
+    autoretry_for=(SMTPServerDisconnected,),
+    default_retry_delay=10,
+    retry_kwargs={'max_retries': 5},
```

Comment:
I think you don't need to conditionally pass this only on retries.
https://docs.celeryproject.org/en/stable/userguide/tasks.html#Task.max_retries

You ought to be able to do:
```suggestion
    max_retries=5,
```